### PR TITLE
gh-98097: Support front() method in queue

### PR DESCRIPTION
**Pull request by mascai**
* Issue: gh-98097

### CODE REVIEW

File: Lib/queue.py [hunk 1]
```diff
@@ -182,6 +182,15 @@ def get(self, block=True, timeout=None):
             self.not_full.notify()
             return item
 
+    def front(self):
+        ''' Get the first element in the queue without removing
+ 
+        If queue is empty, then return None
+        '''
+        if not self.empty():
+            return self.queue[0]
```
Comment (sweeneyde) [inline Lib/queue.py:191]: This has a race condition: a different thread could pop something off of the queue between the execution of these two lines.

Comment (mascai) [reply Lib/queue.py:191]: Should we apply condition variables like this?

```  
    def front(self):
        ''' Get the first element in the queue without removing
 
        If queue is empty, then return None
        '''
        with self.all_tasks_done:
            while self.unfinished_tasks:
                self.all_tasks_done.wait()
                
                if self._qsize():
                    return self.queue[0]
                return None
```

Comment (sweeneyde) [reply Lib/queue.py:191]: You may be able to just do `return self.queue[0]` with `try: ... except IndexError:`. But as I mentioned in the issue, I'm not sure we should be adding such a method at all.

Note also that this has unexpected results for `LifoQueue`, which inherits this method from `Queue`.

**Comment by rhettinger** [comment]
Thanks for the suggestion, but we need to decline because it is incompatible with the guaranteed API for the module which only interacts with the underlying data store via one of four methods:  `_init`, `_qsize`, `_put`, and `_get`.  It is not allowed for `Queue` to access self.queue directly.  Also, there is no guarantee that the underlying object supports indexing.  For example, it would be perfectly reasonable for an implementation to be using a persistent store/load or remote store/load 

Also, even if there had been a way to add compatible this method, we shouldn't do it.  The existing `empty()` and `full()` methods are already lists as being something we will likely remove at some point.  Methods that have a builtin race condition are hard to use correctly and are source of bugs.

**Pull request closed without merging**